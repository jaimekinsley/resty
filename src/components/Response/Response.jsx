import React from 'react';
import ReactJson from 'react-json-view';
import PropTypes from 'prop-types';

const Response = ({ response }) => {
  <>
    <div>
      <ReactJson src={response} />
    </div>
  </>;
};

Response.propTypes = {
  response: PropTypes.oneOfType([PropTypes.object, PropTypes.array]).isRequired
};

export default Response;
