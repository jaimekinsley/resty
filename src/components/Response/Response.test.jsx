import React from 'react';
import { shallow } from 'enzyme';
import Response from './Response';

jest.mock('../services/fetchAPI.js', () => ({
  fetchAPI() {
    return Promise.resolve([
      {
        'character': 'Bender',
        'quote': 'Bite my shiny metal ass.',
        'image': 'https://res.cloudinary.com/dzxqhkyqd/image/fetch/c_scale,w_500/https://res.cloudinary.com/dzxqhkyqd/image/upload/v1552429540/bender.png'
      },
      {
        'character': 'Leela',
        'quote': '...at least here you\'ll be treated with dignity. Now strip naked and get on\nthe probulator.',
        'image': 'https://res.cloudinary.com/dzxqhkyqd/image/upload/v1554904145/leela.png'
      }]);
  }
}));

describe('Response Component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Response />);
  });

  it();
});
