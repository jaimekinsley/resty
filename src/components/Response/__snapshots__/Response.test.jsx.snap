// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Response Component displays a Response object or array 1`] = `
<Fragment>
  <div>
    <t
      collapseStringsAfterLength={false}
      collapsed={false}
      defaultValue={null}
      displayDataTypes={true}
      displayObjectSize={true}
      enableClipboard={true}
      groupArraysAfterLength={100}
      iconStyle="triangle"
      indentWidth={4}
      name="root"
      onAdd={false}
      onDelete={false}
      onEdit={false}
      onSelect={false}
      shouldCollapse={false}
      sortKeys={false}
      src={
        Array [
          Object {
            "character": "Bender",
            "image": "https://res.cloudinary.com/dzxqhkyqd/image/fetch/c_scale,w_500/https://res.cloudinary.com/dzxqhkyqd/image/upload/v1552429540/bender.png",
            "quote": "Bite my shiny metal ass.",
          },
          Object {
            "character": "Leela",
            "image": "https://res.cloudinary.com/dzxqhkyqd/image/upload/v1554904145/leela.png",
            "quote": "...at least here you'll be treated with dignity. Now strip naked and get on
the probulator.",
          },
        ]
      }
      style={Object {}}
      theme="flat"
      validationMessage="Validation Error"
    />
  </div>
</Fragment>
`;
